exercises/say/uSayExample.pas

```diff
@@ -0,0 +1,77 @@
+unit uSay;
+
+interface
+
+type
+  TSay = class
+    class function Say(AInp : Int64) : string;
+  end;
+
+implementation
+
+uses
+  System.SysUtils, System.Math;
+
+{ TSay }
+
+class function TSay.Say(AInp: Int64): string;
+
+const
+  Smalls : Array [0..20] of string = ('', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'ten',
+      'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen', 'seventeen', 'eighteen', 'nineteen', 'twenty');
+  Tens : Array [2..9] of string = ( 'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety');
+  Many : Array [0..4] of string = ('', ' thousand', ' million', ' billion', ' trillion');
+var
+  MC: Integer;
+  S, UT: string;
+
+  function UnderHundred(Ainp : integer) : string;
+  var S : string;
+  begin
+    if Ainp = 0 then
+      exit('');
+    if (Ainp > 20) then
+      Result := format('%s-%s',[Tens[Ainp div 10], Smalls[Ainp mod 10]])
+    else
+      Result := Smalls[Ainp];
+  end;
+
+  function UnderThousand(AInp : integer) : string;
+  var
+    UH, Space: string;
+  begin
+    if AInp = 0 then
+      exit('');
+    UH := UnderHundred(AInp mod 100);
+    if AInp < 100 then
+      Result := UH
+    else
+    begin
+      if UH <> '' then
+        Space := ' ';
+      Result := Format('%s hundred%s%s', [Smalls[AInp div 100], Space, UH]);
```

Comment:
`Space` might not be initialized because of line 50.

Reply:
In Delphi 10.3 you will be able to `var` `Space` and 'UH` like so:
```pascal
  function UnderThousand(AInp : integer) : string;
  begin
    if AInp = 0 then
      exit('');
    var UH: string := UnderHundred(AInp mod 100);
    if AInp < 100 then
      Result := UH
    else
    begin
      var Space: string := '';
      if UH <> '' then
        Space := ' ';
      Result := Format('%s hundred%s%s', [Smalls[AInp div 100], Space, UH]);
    end;
  end;
```
`Space` will then go out of scope after the `end;` after `Result := ...`.  Inline `var`ing and traditional `var`ing is supported, at the same time.  I just changed both variables as an example.